### PR TITLE
Fix Shared Snapshot Cache not Created in Data Path

### PR DESCRIPTION
**Pull request by original-brownbear**
We have to use the node environment instead of the plain environment here to get the
actual data paths.

### CODE REVIEW

File: x-pack/plugin/searchable-snapshots/src/test/java/org/elasticsearch/xpack/searchablesnapshots/AbstractSearchableSnapshotsTestCase.java [hunk 1]
```diff
@@ -163,7 +162,9 @@ protected FrozenCacheService randomFrozenCacheService() {
         if (randomBoolean()) {
             cacheSettings.put(FrozenCacheService.FROZEN_CACHE_RECOVERY_RANGE_SIZE_SETTING.getKey(), randomCacheRangeSize());
         }
-        return new FrozenCacheService(newEnvironment(cacheSettings.build()), threadPool);
+        try (NodeEnvironment nodeEnvironment = newNodeEnvironment()) {
+            return new FrozenCacheService(nodeEnvironment, Settings.EMPTY, threadPool);
```
Comment (ywelsch) [inline x-pack/plugin/searchable-snapshots/src/test/java/org/elasticsearch/xpack/searchablesnapshots/AbstractSearchableSnapshotsTestCase.java:166]: This should use `cacheSettings.build()`

File: x-pack/plugin/searchable-snapshots/src/test/java/org/elasticsearch/xpack/searchablesnapshots/AbstractSearchableSnapshotsTestCase.java [hunk 2]
```diff
@@ -181,16 +182,15 @@ protected CacheService createCacheService(final ByteSizeValue cacheSize, final B
...
-            ),
-            threadPool
-        );
+    protected FrozenCacheService createFrozenCacheService(final ByteSizeValue cacheSize, final ByteSizeValue cacheRangeSize)
+        throws IOException {
+        final Settings settings = Settings.builder()
+            .put(FrozenCacheService.SNAPSHOT_CACHE_SIZE_SETTING.getKey(), cacheSize)
+            .put(FrozenCacheService.SHARED_CACHE_RANGE_SIZE_SETTING.getKey(), cacheRangeSize)
+            .build();
+        try (NodeEnvironment nodeEnvironment = newNodeEnvironment(settings)) {
```
Comment (ywelsch) [inline x-pack/plugin/searchable-snapshots/src/test/java/org/elasticsearch/xpack/searchablesnapshots/AbstractSearchableSnapshotsTestCase.java:191]: no need to create a new node environment here, just use `nodeEnvironment`

**Review by ywelsch — CHANGES_REQUESTED** [review]

File: x-pack/plugin/searchable-snapshots/src/test/java/org/elasticsearch/xpack/searchablesnapshots/AbstractSearchableSnapshotsTestCase.java (hunk 2 above)
Comment (original-brownbear) [reply x-pack/plugin/searchable-snapshots/src/test/java/org/elasticsearch/xpack/searchablesnapshots/AbstractSearchableSnapshotsTestCase.java:191]: ++

File: x-pack/plugin/searchable-snapshots/src/test/java/org/elasticsearch/xpack/searchablesnapshots/AbstractSearchableSnapshotsTestCase.java (hunk 1 above)
Comment (original-brownbear) [reply x-pack/plugin/searchable-snapshots/src/test/java/org/elasticsearch/xpack/searchablesnapshots/AbstractSearchableSnapshotsTestCase.java]: ++

File: x-pack/plugin/searchable-snapshots/src/test/java/org/elasticsearch/xpack/searchablesnapshots/cache/FrozenCacheServiceTests.java [hunk 3]
```diff
@@ -83,11 +85,13 @@ public void testAutoEviction() throws IOException {
...
-        for (Path path : environment.dataFiles()) {
-            Files.createDirectories(path);
-        }
-        try (FrozenCacheService cacheService = new FrozenCacheService(environment, taskQueue.getThreadPool())) {
+        try (
+            NodeEnvironment environment = new NodeEnvironment(settings, TestEnvironment.newEnvironment(settings));
+            FrozenCacheService cacheService = new FrozenCacheService(environment, settings, taskQueue.getThreadPool())
+        ) {
+            for (Path path : environment.nodeDataPaths()) {
+                Files.createDirectories(path);
```
Comment (ywelsch) [inline x-pack/plugin/searchable-snapshots/src/test/java/org/elasticsearch/xpack/searchablesnapshots/cache/FrozenCacheServiceTests.java:93]: This is not necessary anymore

File: x-pack/plugin/searchable-snapshots/src/test/java/org/elasticsearch/xpack/searchablesnapshots/cache/FrozenCacheServiceTests.java [hunk 4]
```diff
@@ -120,11 +124,13 @@ public void testForceEviction() throws IOException {
...
```
Comment (ywelsch) [inline x-pack/plugin/searchable-snapshots/src/test/java/org/elasticsearch/xpack/searchablesnapshots/cache/FrozenCacheServiceTests.java:132]: same here, not necessary

File: x-pack/plugin/searchable-snapshots/src/test/java/org/elasticsearch/xpack/searchablesnapshots/cache/FrozenCacheServiceTests.java [hunk 5]
```diff
@@ -149,11 +155,13 @@ public void testDecay() throws IOException {
...
```
Comment (ywelsch) [inline x-pack/plugin/searchable-snapshots/src/test/java/org/elasticsearch/xpack/searchablesnapshots/cache/FrozenCacheServiceTests.java:163]: same here, not necessary

File: x-pack/plugin/searchable-snapshots/src/test/java/org/elasticsearch/xpack/searchablesnapshots/cache/FrozenCacheServiceTests.java [hunk 6]
```diff
@@ -37,11 +37,13 @@ public void testBasicEviction() throws IOException {
...
```
Comment (ywelsch) [inline x-pack/plugin/searchable-snapshots/src/test/java/org/elasticsearch/xpack/searchablesnapshots/cache/FrozenCacheServiceTests.java:45]: same here, not necessary

File: x-pack/plugin/searchable-snapshots/src/test/java/org/elasticsearch/xpack/searchablesnapshots/AbstractSearchableSnapshotsTestCase.java [hunk 7]
```diff
@@ -163,7 +162,9 @@ protected FrozenCacheService randomFrozenCacheService() {
         if (randomBoolean()) {
             cacheSettings.put(FrozenCacheService.FROZEN_CACHE_RECOVERY_RANGE_SIZE_SETTING.getKey(), randomCacheRangeSize());
         }
-        return new FrozenCacheService(newEnvironment(cacheSettings.build()), threadPool);
+        try (NodeEnvironment nodeEnvironment = newNodeEnvironment()) {
```
Comment (ywelsch) [inline x-pack/plugin/searchable-snapshots/src/test/java/org/elasticsearch/xpack/searchablesnapshots/AbstractSearchableSnapshotsTestCase.java:165]: no need to create new NodeEnvironment here

**Review by ywelsch — APPROVED** [review]
I've left some more comments to address. Once fixed, LGTM

**Comment by original-brownbear** [comment]
Thanks Yannick!

**Pull request merged**